core/transaction_pool/impl/pool_moderator_impl.hpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_POOL_MODERATOR_IMPL_HPP
+#define KAGOME_POOL_MODERATOR_IMPL_HPP
+
+#include <map>
+
+#include "transaction_pool/impl/clock_impl.hpp"
+#include "transaction_pool/pool_moderator.hpp"
+
+namespace kagome::transaction_pool {
+
+  class PoolModeratorImpl : public PoolModerator {
+    static constexpr size_t kDefaultExpectedSize = 2048;
+
+    using SystemClock = Clock<std::chrono::system_clock>;
+    using Map = std::map<common::Hash256, SystemClock::TimePoint>;
+
+   public:
+    /**
+     * @param ban_for amount of time for which a transaction is banned
+     * @param clock a clock used to determine when it is time to unban a
+     * transaction
+     * @param expected_size expected maximum number of banned transactions. If
+     * significantly exceeded, some transactions will be removed from ban list
+     */
+    explicit PoolModeratorImpl(
+        SystemClock::Duration ban_for = std::chrono::minutes(30),
+        std::shared_ptr<SystemClock> clock =
+            std::make_shared<kagome::transaction_pool::SystemClock>(),
+        size_t expected_size = kDefaultExpectedSize);
```

Comment:
expected size and ban_for should be in a separate struct, maybe called `PoolModeratorConfig`, and then used here as last arg.

Reply:
I like that idea, but I'd prefer to use it as the first arg, because I think that it would be more often used than clock.

Reply:
All components **will not** be created manually, only through [boost].DI, so this does not matter. There should not be any default args, only for logger and config.

Reply:
Okay, I've got no experience with BoostDI, so I will just rely on your words.